src/Toolkit/Toolkit.Maui/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -47,6 +50,31 @@ static SymbolDisplay()
     public SymbolDisplay()
     {
         ControlTemplate = DefaultControlTemplate;
+        this.PropertyChanged += SymbolDisplay_PropertyChanged;
+    }
+
+    private void SymbolDisplay_PropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(Window) && Window != null)
+        {
+            if (_lastScaleFactor != GetScaleFactor())
+            {
+                Refresh();
+            }
+
+            _displayDensityChangedListener = new WeakEventListener<Window, object, DisplayDensityChangedEventArgs>(Window)
```

Comment:
Add `_displayDensityChangedListener?.Detach()` before creating a new one, which ensures immediate and full cleanup of the previous listener, and avoid risk of firing multiple times.